nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -95,6 +95,54 @@ void VoxelLayer::onInitialize()
 
   unknown_threshold_ += (VOXEL_BITS - size_z_);
   matchSize();
+  setParamCallbacks();
+}
+
+void
+VoxelLayer::setParamCallbacks()
```

Comment:
All these layers need the mutex to be locked before modifying resources asynchronously.